dask/dataframe/io/io.py

```diff
@@ -646,24 +650,25 @@ def to_records(df):
     return df.map_partitions(M.to_records)
 
 
-# TODO: type this -- causes lots of papercuts
 @insert_meta_param_description
 def from_delayed(
-    dfs,
+    dfs: Union[Delayed, Iterable[Union[Delayed, distributed.Future]]],
     meta=None,
-    divisions=None,
-    prefix="from-delayed",
-    verify_meta=True,
+    divisions=Union[tuple, Literal["sorted"], None],
```

Comment:
```suggestion
    divisions: tuple[Any, ...] | Literal["sorted"] | None = None,
```
The `tuple[Any, ...]` may be able to be tightened but I'm not sure off the top of my head.

Reply:
So at least `str | int` are possible divisions. Are there any other possibilities?

Reply:
Yeah, datetimes are also possible. I'm not sure that we can do much better than `Any`, since there might also be extension dtypes.